docs/create-a-ui-component.md

```diff
@@ -0,0 +1,280 @@
+
+## How to build a UI Component?
+
+Alright, that's nice in theory, but how does it translate in practice? We'll try
+to get a bit more tangible by creating a UI component that we would need in a
+Reinsurance Banner.
+
+Image de ce qu'on va construire.  
+Mockup of the reinsurance banner that we will implement
+
+### Step 1: Defining the components
+
+First, let's split the mockup in several UI components.
+
+Image avec les zones
+
+* **`atoms/Heading`:** enforce consistent font-sizes in our theme for any title
+* **`atoms/Icon`:** enforces icon sizes and accessibility guidelines
+* **`molecules/MediaCard`:** displays a Card described by an image and aligns
+  images properly accross the whole theme
+* **`organisms/InlineCardList`:** manages a list of cards and inlines them,
+  regardeless of the device size.
+
+> If you have troubles splitting your mockups, you can refer to
+> [Thinking in React](https://reactjs.org/docs/thinking-in-react.html) in the
+> official React documentation or to
+> [Brad Frost's book about Atomic Design](http://atomicdesign.bradfrost.com/).  
+> Additionnally, the way we splitted things here depends on what you've already built,
+> you may organize your code differently and that's perfectly fine. Especially since,
+> we've splitted the components **a lot** here. It may be a better idea not to go
+> overboard. It's often easier to wait and see how it can be reused later.
+
+We won't be able to detail each component here. We will focus on
+`molecules/MediaCard` instead. But keep in mind that any UI component in
+Front-Commerce will look similar to what we are going to build here.
+
+### Step 2: Setup your dev environment
+
+Before doing the actual work let's bootstrap our dev environment. To do so, we
+will need to create three files:
+
+* `src/web/theme/ui/molecules/MediaCard/MediaCard.js`: will be your actual
+  component
+
+  ```jsx
+  import React from "react";
+
+  const MediaCard () => <div>Media Card</div>;
+
+  MediaCard.propTypes = {
+      // Don't forget to setup your PropTypes
+      // here since this component will be heavily
+      // used through your application
+  }
+
+  export default MediaCard
+  ```
+
+* `src/web/theme/ui/molecules/MediaCard/index.js`: will only proxy the
+  MediaCard.js file in order to be able to do imports on the folder directly.
+
+  ```jsx
+  import MediaCard from "./MediaCard.js";
+
+  export default MediaCard;
+  ```
+
+* `src/web/theme/ui/molecules/MediaCard/MediaCard.story.js`: will add a story to
+  the [Storybook](https://storybook.js.org/) of your application. This will
+  serve as living documentation and will allow anyone to easily understand what
+  is MediaCard used for and how to use it. Learn more about our Storybook usage
+  in (#TODO-documentation).
+
+  ```jsx
+  import MediaCard from "./MediaCard.js";
+  import { storiesOf } from "@storybook/react";
+
+  storiesOf("molecules.MediaCard", module).add("default", () => {
+    return <MediaCard />;
+  });
+  ```
+
```

Comment:
It would be great to mention and recommend to (optionally) add a `readme.md` to document the component’s purpose here.

Also, a directory tree list either at the beginning or at the end may help to visualize the structure better.